crypto/default/src/test/java/org/keycloak/crypto/def/test/DefaultSecureRandomTest.java

```diff
@@ -0,0 +1,40 @@
+package org.keycloak.crypto.def.test;
+
+import java.security.SecureRandom;
+
+import org.jboss.logging.Logger;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.keycloak.common.crypto.CryptoIntegration;
+import org.keycloak.rule.CryptoInitRule;
+
+/**
+ * @author <a href="mailto:mposolda@redhat.com">Marek Posolda</a>
+ */
+public class DefaultSecureRandomTest {
+
+    @ClassRule
+    public static CryptoInitRule cryptoInitRule = new CryptoInitRule();
+
+    protected static final Logger logger = Logger.getLogger(DefaultSecureRandomTest.class);
+
+    @Test
+    public void testSecureRandom() throws Exception {
+        logger.info(CryptoIntegration.dumpJavaSecurityProviders());
+
+        SecureRandom sc1 = new SecureRandom();
+        logger.infof(dumpSecureRandom("new SecureRandom()", sc1));
+
+        SecureRandom sc3 = SecureRandom.getInstance("SHA1PRNG");
+        logger.infof(dumpSecureRandom("SecureRandom.getInstance(\"SHA1PRNG\")", sc3));
```

Comment:
Shouldn't this be some asserts instead of logging?

Reply:
Good point. I've added some asserts for the things, which are safe to test on all platforms (EG. for `SecureRandom.getInstance()` is not possible to test anything as both default provider and algorithm are dependent on the platform where the test is executed).

I've originally used unit tests just to test on various platforms that SecureRandom is possible to instantiate (no exception thrown) for the given algorithm.